### ISSUE TITLE
Cannot type into field

### ISSUE THREAD

**Issue by hypeofpipe** [cypress-io/cypress#1528]
Operating System: Solus 3.999 shannon
Cypress Version: 2.1.0
Browser Version: Electron 59

Is this a Feature or Bug?
This is a bug.

Current behavior:

I have an input field:
<input name="passengers[0].birth_date" class="FieldGroup__customFieldSm___12eZY" placeholder="DD-MM-YYYY" value="" type="">

I get the error:

> CypressError: cy.type() can only be called on textarea or :text. Your subject is a: <input type="" name="passengers[0].birth_date" class="FieldGroup__customFieldSm___12eZY" placeholder="DD-MM-YYYY" value="">

Desired behavior:

This is an input element, so I would expect the input to be entered.

How to reproduce:
Create an input, and try type().

Test code:

```js
    cy.get(selector)
        .click()

    cy.get(selector)
        .type(02011999 + '{enter}')
```

**Comment by jennifer-shehane** [comment]
Can you paste the HTML code of the input you're trying to type into?

**Comment by hypeofpipe** [comment]
@jennifer-shehane  oh yeah, I've completely forgot it :)

```
<div class="Checkout__fieldRow___vzW2S">
   <div class="Checkout__labelCol___2Y_f1"><label for="birth_date">Date of birth</label></div>
   <div class="Checkout__fieldCol___1zvrt">
      <div class="FieldGroup__DatePickerGroup___3BZyo form-group"><input name="passengers[0].birth_date" class="FieldGroup__customFieldSm___12eZY" placeholder="DD-MM-YYYY" value=""><label for="passengers[0].birth_date" class="FieldGroup__placeholder___3SZMc">DD-MM-YYYY</label></div>
   </div>
</div>
```

**Comment by jennifer-shehane** [comment]
Yeah, so I believe that this error is displaying because you haven't defined a `type` attribute on your input. 

Reading the HTML spec, they suggest that if no `type` exists, that it should be treated as `type='text'`. We should just be treating any input without a `type` as `text` automatically. - https://html.spec.whatwg.org/multipage/input.html#the-input-element

> The missing value default and the invalid value default are the `Text` state.

**Workaround:** Add a `type` attribute to your input.

**Permanent Fix:** Cypress should treat any input with no `type` attribute defined as `type` text. 
- I believe the `isTextLike` method should be expanded to include `<input>` elements with no `type` attribute here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/dom/elements.coffee#L74
- You could also potentially expand the checks for throwing the `not_on_text_field` error on `.type()` to just do this check, since I'm not exactly sure what else changing `isTextLike` would affect. https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/actions/type.coffee#L93

**Comment by rajdhandus** [comment]
Hello.. can I please work on this issue?

**Comment by jennifer-shehane** [comment]
Ok, so I probably should have tested my theory first before rambling off a solution. The missing `type` attribute does not appear to be the issue here, as I can type into an input without a type attribute fine in Cypress. 

@hypeofpipe Could you tell me what `selector` is it you were using in your original code? I realize that you didn't include everything needed to replicate.

**Comment by hypeofpipe** [comment]
@jennifer-shehane It is fixed by adding a type to it. Nevertheless, I think Cypress should give the opportunity to type in those fields (with a null type).

**Comment by Bkucera** [comment]
This should be fixed as of 3.0.3 along with #1241, as we read of the `type` property instead of the attribute
